build/package/install_agent_spawn_host.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -ex
+
+while getopts 'i:h:t:u:v:' opt; do
+    case $opt in
+        i) keyfile="$OPTARG";;      # SSH identity file
+        u) user="$OPTARG";;         # Username on the remote host
+        h) hostsfile="$OPTARG";;    # Output of Evergreen host.list
+        t) groupid="$OPTARG";;      # APIGroupId
+        v) apiKey="$OPTARG";;       # APIKey
+        *) exit 1;;
+    esac
+done
+
+hosts=$(cat << EOF | python - "$hostsfile"
+import sys
+import json
+with open(sys.argv[1]) as hostsfile:
+    hosts = json.load(hostsfile)
+    for host in hosts:
+        print(host["dns_name"])
+EOF
+)
+for host in $hosts; do
+    set +e
+    echo "Installing dependeces on $host"
+    ssh -i "$keyfile" -o ConnectTimeout=50 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -tt "$user@$host" 'bash -s' <<'ENDSSH'
+        free -m
+        sudo apt-get install -y --no-install-recommends ca-certificates curl logrotate openssl snmp && exit
+ENDSSH
+    echo "Installing the automation agent on $host"
+    ssh -i "$keyfile" -o ConnectTimeout=50 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -tt "$user@$host" ARG1="$groupid" ARG2="$apiKey" 'bash -s' <<'ENDSSH'
+        free -m
+        echo "Downloadind and extracting the automation agent"
+        curl -OL https://cloud-dev.mongodb.com/download/agent/automation/mongodb-mms-automation-agent-manager_10.15.0.6409-1_amd64.ubuntu1604.deb
+        sudo dpkg -i mongodb-mms-automation-agent-manager_10.15.0.6409-1_amd64.ubuntu1604.deb
+
+        echo "Replacing mmsGroupId and mmsApiKey properties"
+        sudo sed -i "s/\(mmsGroupId *= *\).*/\1$ARG1/" /etc/mongodb-mms/automation-agent.config
+        sudo sed -i "s/\(mmsApiKey *= *\).*/\1$ARG2/" /etc/mongodb-mms/automation-agent.config
+
+        echo "Preparing the /data directory to store your MongoDB data"
+        sudo mkdir -p /data
+        sudo chown mongod:mongod /data
+
+        echo "Starting the agent"
+        sudo systemctl start mongodb-mms-automation-agent.service
+        exit
```

Comment:
you can create another bash file you can reference here, this would make this more maintainable
like, having a `agent-config.sh` file with this and then here do something like 
`ssh ... 'bash ~/agent-config.sh`

Reply:
I tried to do it but I got  `No such file or directory` and it seems that it is looking for that file inside the ubuntu 16.04 spawn host.

Reply:
I googled this and It seems that running a local script in the ssh host is not so easy when there are arguments to pass.